opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -94,8 +95,11 @@ public void finish() {
 
     @Override
     public synchronized void finish(long finishMicros) {
-        this.finishMicros = finishMicros;
-        this.mockTracer.appendFinishedSpan(this);
+        if (!finished) {
```

Comment:
since MockSpan is mainly used for testing, AND since it's actually undefined behavior to call `finish` multiple times on a single Span instance, I'd argue that this should at least log something in the `finished==true` case. thoughts?